src/App.tsx

```diff
@@ -1,41 +1,36 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import 'bulma/css/bulma.css';
+import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
-import './styles/general.scss';
-import { PostsList } from './components/PostsList';
-import { PostDetails } from './components/PostDetails';
 
-const App: React.FC = () => (
-  <div className="App">
-    <header className="App__header">
-      <label>
-        Select a user: &nbsp;
+import { UserSelector } from './components/UserSelector';
+import { User } from './types/User';
+import { getUsers } from './utils/fetchFunctions';
+import { MainContent } from './components/MainContent';
+import { Sidebar } from './components/Sidebar';
 
-        <select className="App__user-selector">
-          <option value="0">All users</option>
-          <option value="1">Leanne Graham</option>
-          <option value="2">Ervin Howell</option>
-          <option value="3">Clementine Bauch</option>
-          <option value="4">Patricia Lebsack</option>
-          <option value="5">Chelsey Dietrich</option>
-          <option value="6">Mrs. Dennis Schulist</option>
-          <option value="7">Kurtis Weissnat</option>
-          <option value="8">Nicholas Runolfsdottir V</option>
-          <option value="9">Glenna Reichert</option>
-          <option value="10">Leanne Graham</option>
-        </select>
-      </label>
-    </header>
+export const App = () => {
+  const [users, setUsers] = useState<User[]>([]);
 
-    <main className="App__main">
-      <div className="App__sidebar">
-        <PostsList />
-      </div>
+  useEffect(() => {
+    getUsers().then(setUsers);
+  }, []);
 
-      <div className="App__content">
-        <PostDetails />
+  return (
+    <main className="section">
+      <div className="container">
+        <div className="tile is-ancestor">
+          <div className="tile is-parent">
+            <div className="tile is-child box is-success">
+              <div className="block">
+                <UserSelector users={users} />
```

Comment:
Make sure you described objects in propTypes for the UserSelector component to ensure type checking.